Tests/MessagingPush/MessagingPushTest.swift

```diff
@@ -0,0 +1,156 @@
+            expect.fulfill()
+        }
+        
+        waitForExpectations()
+    }
+    
+    func test_registerDeviceToken_givenHttpSuccess_expectSaveExpectedData() {
+        
+        identifyRepositoryMock.setIdentifierClosure = { identifier in
+            self.identifyRepositoryMock.identifier = identifier
+        }
+        
+        mockCustomerIO.setIdentifier(identifier: String.random, onComplete: { result in
+            guard case .success = result else { return XCTFail() }
+            XCTAssertNotNil(self.mockCustomerIO.identifier)
+        })
+        
+        httpClientMock.requestClosure = { params, onComplete in
+            onComplete(Result.success(Data()))
+        }
+        
+        let actualToken = Data(String.random.utf8)
+        
+        let expect = expectation(description: "Expect to persist token")
+        self.messagingPush.registerDeviceToken(deviceToken: actualToken) { result in
+            guard case .success = result else { return XCTFail() }
+            expect.fulfill()
+        }
+        
+        guard let storedToken = self.messagingPush.deviceToken else {
+            return XCTFail()
+        }
+        
+        XCTAssertEqual(storedToken, actualToken)
+
+        waitForExpectations()
+    }
+    
+    // MARK: deleteDeviceToken
+    
+    func test_deleteDeviceToken_expectFailIfNotIdentified() {
+                
+        httpClientMock.requestClosure = { params, onComplete in
+            onComplete(Result.success(Data()))
+        }
+        
+        let expect = expectation(description: "Expect to fail to delete device token")
+        self.messagingPush.deleteDeviceToken() { result in
+            print(result)
+            guard case .failure(let error) = result else { return XCTFail() }
+            guard case .noCustomerIdentified = error else { return XCTFail() }
+            expect.fulfill()
+        }
+
+        waitForExpectations()
+    }
+    
+    func test_deleteDeviceToken_expectFailIfNoToken() {
+        
+        identifyRepositoryMock.setIdentifierClosure = { identifier in
+            self.identifyRepositoryMock.identifier = identifier
+        }
+        
+        mockCustomerIO.setIdentifier(identifier: String.random, onComplete: { result in
+            guard case .success = result else { return XCTFail() }
+            XCTAssertNotNil(self.mockCustomerIO.identifier)
+        })
+                
+        httpClientMock.requestClosure = { params, onComplete in
+            onComplete(Result.success(Data()))
+        }
+        
+        let expect = expectation(description: "Expect to fail to delete device token")
+        self.messagingPush.deleteDeviceToken() { result in
+            guard case .failure(let error) = result else { return XCTFail() }
+            guard case .deviceNotRegistered = error else { return XCTFail() }
+            expect.fulfill()
+        }
+
+        waitForExpectations()
```

Comment:
Good to also make sure the http mock is never called. Something like: `AssertFalse(httpclientmock.requestcalled)` to make sure our SDK when it errors does not try to perform the action of the function.